### PR TITLE
More development dashboards

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Organizes json files into folder structure (on disk not in Grafana)
* [x] Aligns tags across all so we can filter based on type of dashboard
* [x] Adds new home dashboard with a bunch of dashboard list with filters for different types of dashboards
* [x] Adds repeating panels test dashboard (using Test Data datasource)
* [x] Adds a basic demo dashboard (that uses test data)
* [x] Adds Bar Gauge demo dashboards

While working on this I came up with a provisioning enhancement. Show ExternalId (path on disk) in the save modal for provisioned dashboards. 

thoughts @bergquist ^

The home / overview dashboard:
image

### CODE REVIEW

File: devenv/dashboards.yaml [hunk 1]
```diff
@@ -4,6 +4,6 @@ providers:
  - name: 'gdev dashboards'
    folder: 'gdev dashboards'
    type: file
-   updateIntervalSeconds: 15
+   updateIntervalSeconds: 60
```
Comment (bergquist) [inline devenv/dashboards.yaml]: wont this just be frustrating when working with dashboards?
I think all devs runs SSD nowdays and the content should almost always be cached. scanning a few files are very very very fast :)

**Comment by bergquist** [comment]
Interesting idea. But couldn't that be considered information leakage? 
But it would be nice in devenv.

File: devenv/dashboards.yaml (hunk 1 above)
Comment (torkelo) [reply devenv/dashboards.yaml]: yea I increased it a bit after testing with many folders defined here (each have it's own interval) and the log spamming was a bit too much. And if we end up with a lot of dashboards parsing all those json files is going to eat up a lot cpu. The way the scanning works provisioning update cycles are far from cheap, having to deserialize all dashboards, and to hashed checksum on content, at least. 

Set to 3s with 1300 dashboards it eats 100% according to this user: https://github.com/grafana/grafana/issues/16118

**Comment by torkelo** [comment]
> Interesting idea. But couldn't that be considered information leakage?

Sort of, especially if we include the full path (which ExternalId does contain) so you would see where on disk dashboards files are located. So we could strip out the first parts and only paths after the root provisioned path (but the logic for that could be a bit tricky). 

Just think the updating a provisioned dashboard step is still pretty complicated as there is currently no hints on what file I need to update. A pretty major oversight in the workflow.

File: devenv/dashboards.yaml (hunk 1 above)
Comment (torkelo) [reply devenv/dashboards.yaml]: But that 100% could be due to something else, maybe huge > megabytes big json dashboards (can happen)

**Comment by bergquist** [comment]
yea your absolutly right.
I wonder if getting the relative path would be helpful enough. We can certainly make that possible.

**Review by bergquist — APPROVED** [review]
LGTM!

I opened https://github.com/grafana/grafana/issues/16587 about show path to provisioned dashboard

**Pull request merged**